From e5fe2cbebcb666bb132ab26d8dacbace383936c7 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 07:40:41 +0200
Subject: [PATCH] avformat/rtpdec_h264: Ignore framesize attributes without a
 height

Fixes: out of array read
Fixes: xfnEsbsbvEIX
Fixes: AISLE-2026-0111-00033
Stack out of array read Replicated through UnModified FFmpeg (ffprobe) with ASAN
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
---
 libavformat/rtpdec_h264.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/rtpdec_h264.c b/libavformat/rtpdec_h264.c
index 72724c290977..c145edaa5e59 100644
--- a/libavformat/rtpdec_h264.c
+++ b/libavformat/rtpdec_h264.c
@@ -198,6 +198,9 @@ void ff_h264_parse_framesize(AVCodecParameters *par, const char *p)
         *dst++ = *p++;
     *dst = '\0';
 
+    if (*p != '-')
+        return;
+
     // a='framesize:96 320-240'
     // set our parameters
     par->width   = atoi(buf1);
